lua/starfall/libs_sh/mesh.lua

```diff
@@ -762,6 +775,191 @@ if CLIENT then
+	-- @param position Vector
+	-- @client
+	function mesh_library.writePosition(pos)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.Position, {vunwrap(pos)}})
+	end
+	
+	--- Sets the vertex texture coordinates
+	-- @param stage Number, stage of the texture coordinate
+	-- @param u Number, u coordinate
+	-- @param v Number, v coordinate
+	-- @client
+	function mesh_library.writeUV(stage, u, v)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		checkluatype(stage, TYPE_NUMBER)
+		checkluatype(u, TYPE_NUMBER)
+		checkluatype(v, TYPE_NUMBER)
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.TexCoord, {stage, u, v}})
+	end
+	
+	--- Sets the vertex user data
+	-- @param tangent_x
+	-- @param tangent_y
+	-- @param tangent_z
+	-- @param tangent_handedness
+	-- @client
+	function mesh_library.writeUserData(x, y, z, handedness)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		checkluatype(x, TYPE_NUMBER)
+		checkluatype(y, TYPE_NUMBER)
+		checkluatype(z, TYPE_NUMBER)
+		checkluatype(handedness, TYPE_NUMBER)
+		
+		meshd.modified_vertex = true
+		table.insert(meshd.instructions, {mesh.UserData, {x, y, z, handedness}})
+	end
+	
+	--- Draws a quad using 4 vertices
+	-- @client
+	function mesh_library.writeQuad(v1, v2, v3, v4)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		meshd.modified_vertex = true
+		meshd.true_vertices = meshd.true_vertices + 4
+		table.insert(meshd.instructions, {mesh.Quad, {vunwrap(v1), vunwrap(v2), vunwrap(v3), vunwrap(v4)}})
+	end
+	
+	--- Draws a quad using a position, normal and size
+	-- @param position Vector
+	-- @param normal Vector
+	-- @param w Number
+	-- @param h Number
+	-- @client
+	function mesh_library.writeQuadEasy(pos, norm, w, h)
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		checkluatype(w, TYPE_NUMBER)
+		checkluatype(h, TYPE_NUMBER)
+		
+		meshd.modified_vertex = true
+		meshd.true_vertices = meshd.true_vertices + 4
+		table.insert(meshd.instructions, {mesh.QuadEasy, {vunwrap(pos), vunwrap(norm), w, h}})
+	end
+	
+	--- Pushes the vertex data onto the render stack
+	-- @client
+	function mesh_library.advanceVertex()
+		local meshd = instance.data.mesh_dynamic
+		if not meshd.started then SF.Throw("Dyanic mesh not started.", 2) end
+		
+		if meshd.modified_vertex then
+			meshd.vertices = meshd.vertices + 1
+			meshd.true_vertices = meshd.true_vertices + 1
+			meshd.modified_vertex = false
+		end
+		
+		table.insert(meshd.instructions, {mesh.AdvanceVertex, {}})
+	end
+	
+	--- Returns the amount of vertices that have yet been pushed
```

Comment:
--- Returns the amount of vertices that have been pushed

yet seems redundant and weird, I am not native speaker though

Reply:
Same as https://wiki.facepunch.com/gmod/mesh.VertexCount